src/hotspot/share/prims/jvmtiExport.cpp

```diff
@@ -1560,31 +1560,19 @@ void JvmtiExport::post_method_entry(JavaThread *thread, Method* method, frame cu
   }
 }
 
-void JvmtiExport::post_method_exit(JavaThread *thread, Method* method, frame current_frame) {
+void JvmtiExport::post_method_exit(JavaThread* thread, Method* method, frame current_frame) {
   HandleMark hm(thread);
   methodHandle mh(thread, method);
 
-  EVT_TRIG_TRACE(JVMTI_EVENT_METHOD_EXIT, ("[%s] Trg Method Exit triggered %s.%s",
-                     JvmtiTrace::safe_get_thread_name(thread),
-                     (mh() == NULL) ? "NULL" : mh()->klass_name()->as_C_string(),
-                     (mh() == NULL) ? "NULL" : mh()->name()->as_C_string() ));
-
   JvmtiThreadState *state = thread->jvmti_thread_state();
-  if (state == NULL || !state->is_interp_only_mode()) {
-    // for any thread that actually wants method exit, interp_only_mode is set
-    return;
-  }
-
   // return a flag when a method terminates by throwing an exception
   // i.e. if an exception is thrown and it's not caught by the current method
   bool exception_exit = state->is_exception_detected() && !state->is_exception_caught();
```

Comment:
So this only applies to the case where the post_method_exit comes from remove_activation?  Good to have it only on this path in this case.

Reply:
I'm not sure. There might be other cases, when remove_activation is called by the exception code. That's why I didn't want to change it to just true in this path.

Reply:
The post_method_exit can come from Zero interpreter:
```
src/hotspot/share/interpreter/zero/bytecodeInterpreter.cpp:            
     CALL_VM_NOCHECK(InterpreterRuntime::post_method_exit(THREAD));
```

Reply:
It seems like the case of exception_exit is a condition from the call from notice_unwind_due_to_exception() and not the calls from InterpreterRuntime::post_method_exit() for both callers.  But maybe if the exception is set and not caught, this post_method_exit() is called when unwinding to the exception handler.  I can't tell, so leave it to be safe.